client/src/components/graphs/OrganizationalChart.js

```diff
@@ -69,15 +72,34 @@ const transitionDuration = 200
 
 const ranks = Settings.fields.person.ranks.map(rank => rank.value)
 
-const sortPositions = (positions, truncateLimit) => {
-  const allResults = [...positions].sort((p1, p2) =>
-    ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank) ? -1 : 1
-  )
+const leaderCountMap = {}
+
+const sortPositions = (uuid, positions, truncateLimit) => {
+  const allResults = [...positions]
+    .sort((p1, p2) => {
+      if (p1.role === p2.role) {
+        return ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank)
+          ? -1
+          : 1
+      }
+      return Object.keys(PositionRole).indexOf(p1.role) >
+        Object.keys(PositionRole).indexOf(p2.role)
+        ? -1
+        : 1
+    })
+    .map(pos => ({ ...pos, orgUuid: uuid }))
```

Comment:
You're creating new objects here every time this function is called (which is a lot). It is probably much simpler to also request `organization { uuid }` in the GraphQL query, and refer to `d.organization.uuid` instead of `d.orgUuid` where you need it. Then you can just remove this mapping.